rtengine/imageio.cc

```diff
@@ -1621,3 +1621,57 @@ int ImageIO::save (Glib::ustring fname)
         return IMIO_FILETYPENOTSUPPORTED;
     }
 }
+
+void ImageIO::setProgressListener (ProgressListener* l)
+{
+    pl = l;
+}
+
+void ImageIO::setSampleFormat(IIOSampleFormat sFormat)
+{
+    sampleFormat = sFormat;
+}
+
+IIOSampleFormat ImageIO::getSampleFormat() const
+{
+    return sampleFormat;
+}
+
+void ImageIO::setSampleArrangement(IIOSampleArrangement sArrangement)
+{
+    sampleArrangement = sArrangement;
+}
+
+IIOSampleArrangement ImageIO::getSampleArrangement() const
+{
+    return sampleArrangement;
+}
+
+cmsHPROFILE ImageIO::getEmbeddedProfile () const
+{
+    return embProfile;
+}
+
+void ImageIO::getEmbeddedProfileData (int& length, unsigned char*& pdata) const
+{
+    length = loadedProfileLength;
+    pdata = (unsigned char*)loadedProfileData;
+}
+
+MyMutex& ImageIO::mutex ()
+{
+    return imutex;
```

Comment:
Aha, now I get it. Crappy name. How about `MyMutex& ImageIO::getMutex() const`?

Reply:
Sorry, can't be `const` when returning a non-const reference. Missed that...

Reply:
Scratch my last comment